### PR TITLE
make find-references behave better with simpler code

### PR DESCRIPTION
**Pull request by kermorgant**
ensime-mode has a similar implementation, which does not rely on
tabulated mode and I liked the shorter implementation (+ grep-mode makes
it easy to kill the buffer with a single keystroke).

Besides, set-window-dedicated-p helps make the buffer sticky.

partly solves #98

### CODE REVIEW

**Comment by kermorgant** [comment]
I might give it another stab current appearance of ref list is not as good as with tabulated mode

File: phpactor.el [hunk 1]
```diff
@@ -356,38 +356,27 @@ of GitHub.")
...
-      (tabulated-list-init-header)
-      (tabulated-list-print t)
-      (switch-to-buffer-other-window phpactor-references-buffer))))
+    (switch-to-buffer (get-buffer-create phpactor-references-buffer))
+    (set-window-dedicated-p (get-buffer-window) t)
+    (setq buffer-read-only nil)
+    (erase-buffer)
+    (dolist (file-reference current-references)
+      (let ((path (plist-get file-reference :file))
+            (x nil))
```
Comment (zonuexe) [inline phpactor.el]: Is `x` unused?

File: phpactor.el [hunk 2]
```diff
@@ -356,38 +356,27 @@ of GitHub.")
...
+        (dolist (reference (plist-get file-reference :references))
+          (insert (number-to-string (plist-get reference :line_no)))
+          (insert ": ")
+          (when path
+            (insert-text-button (phpactor-truncate-left path phpactor-references-list-col1-width)
+                        'action (lambda(x) (find-file path (plist-get reference :start)))
+                        'help-echo "mouse-2: visit this file in other window")
```
Comment (zonuexe) [inline phpactor.el]: * If you ignore the argument, the variable name should be `_`.
* Please indent this line up properly.
* Separate between `lambda` and the argument list with a space.

**Comment by zonuexe** [comment]
I commented on details, but it looks good to me.

**Comment by kermorgant** [comment]
@zonuexe 
I added goto-char instruction, reordered the "cols" (more like the initial tabulated mode) and fixed the things you mentioned (except maybe the indent ?)

**Pull request merged**